pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -939,6 +942,31 @@ type NomadSDConfig struct {
 	// +optional
 	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
 }
+
+// OVHCloudSDConfig configurations allow retrieving scrape targets from OVHcloud's dedicated servers and VPS using their API.
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ovhcloud_sd_config
+// +k8s:openapi-gen=true
+type OVHCloudSDConfig struct {
+	// Access key to use. https://api.ovh.com.
+	// +required
+	ApplicationKey string `json:"applicationKey"`
+	// +required
+	ApplicationSecret *corev1.SecretKeySelector `json:"applicationSecret"`
+	// +required
+	ConsumerKey *corev1.SecretKeySelector `json:"consumerKey"`
+	// Service of the targets to retrieve. Must be `vps` or `dedicated_server`.
+	// +kubebuilder:validation:Enum=vps;dedicated_server
+	// +optional
+	Service *string `json:"service,omitempty"`
+	// Custom endpoint to be used.
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	Endpoint *string `json:"endpoint,omitempty"`
```

Comment:
hmm looks like Endpoint is a required field.

```suggestion
	// +required
	Endpoint string `json:"endpoint"`
```

Reply:
Got it. Done